### PR TITLE
HDDS-6959. Fix move timeout in latest iteration metric

### PR DESCRIPTION
**Pull request by kerneltime**
## What changes were proposed in this pull request?

Fix move timeout in latest iteration metric

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-6959

## How was this patch tested?
Added a basic unit test

### CODE REVIEW

**Comment by kerneltime** [comment]
> @kerneltime Thanks for working on this. I think we need some refactoring here. With this change, post move book-keeping will be done at three different places:
> 
> 1. `whenComplete((result, ex) -> {...}` in `ContainerBalancer#moveContainer`
This gets called when the underlying replication manager completes
> 2. `ContainerBalancer#checkIterationMoveResults`
This gets evaluated for timeout when the logic in ContainerBalancer gives up on the replication.
> 3. `ContainerBalancerMetrics#incrementCurrentIterationContainerMoveMetric` which is called in `ContainerBalancer#moveContainer`
This is the same as 1. 
> 
> Do you think it's better to update metrics at just one place, preferably `ContainerBalancer#checkIterationMoveResults`?

There are 2 different ways by which a timeout event can occur. We could split the timeout into 2 separate counters based on who gave up but that should be a different PR.

There are 2 tests in `TestContainerBalancer` which trigger different ways in which timeout can occur and increment the same counter.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancerMetrics.java [hunk 1]
```diff
@@ -121,6 +122,42 @@ public void incrementNumContainerMovesCompletedInLatestIteration(
...
+    case REPLICATION_FAIL_INFLIGHT_REPLICATION:
+    case FAIL_NOT_LEADER:
+    case REPLICATION_FAIL_NOT_EXIST_IN_SOURCE:
+    case REPLICATION_FAIL_EXIST_IN_TARGET:
+    case REPLICATION_FAIL_CONTAINER_NOT_CLOSED:
+    case REPLICATION_FAIL_INFLIGHT_DELETION:
+    case REPLICATION_FAIL_NODE_NOT_IN_SERVICE:
+    case DELETION_FAIL_NODE_NOT_IN_SERVICE:
+    case REPLICATION_FAIL_NODE_UNHEALTHY:
+    case DELETION_FAIL_TIME_OUT:
```
[the remark this answers is not in the document]
Comment (kerneltime) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancerMetrics.java]: Makes sense.

**Review by siddhantsangwan — APPROVED** [review]
Looks good!

**Comment by adoroszlai** [comment]
Thanks @kerneltime for the fix, @siddhantsangwan for the review.

**Pull request merged**